data/spells/scripts/attack/envenom.lua

```diff
@@ -6,8 +6,8 @@ combat:setParameter(COMBAT_PARAM_DISTANCEEFFECT, CONST_ANI_EARTH)
 function onTargetCreature(creature, target)
 	local min = (creature:getLevel() / 80) + (creature:getMagicLevel() * 0.55) + 6
 	local max = (creature:getLevel() / 80) + (creature:getMagicLevel() * 0.75) + 7
-	local damage = math.random(math.floor(min), math.floor(max))
-	creature:addDamageCondition(target, CONDITION_POISON, 1, target:isPlayer() and math.floor(damage / 2 + 0.5) or damage)
+	local damage = math.random(math.floor(min) * 1000, math.floor(max) * 1000) / 1000
```

Comment:
The values of ``min`` and ``max`` are the values generated by the min and max formulas. As a way of keeping the definition of min and max formulas cleaner, I chose not to use ``floor`` on the same line, and instead I decided to go for:
```lua
local damage = math.random(math.floor(min), math.floor(max))
```
instead of:
```lua
local min = math.floor((creature:getLevel() / 80) + (creature:getMagicLevel() * 0.55)) + 6
local max = math.floor((creature:getLevel() / 80) + (creature:getMagicLevel() * 0.75)) + 7
-- then
local damage = math.random(min, max)
```

So we should not use the values generated by the min and max formulas as a way of acquiring such precision.

So, now we get two integer values from min and max using ``floor`` on ``math.random(math.floor(min), math.floor(max))``, but this logarithmic formula requires atleast 3 digits of precision to have the behavior we're looking for for such damage conditions (bleeding, and poison originally), so I multiply both min and max integer values by 1000 and then I divided that by 1000 (permille).

While it is true that both min and max may sometimes generate numbers with 3 digits of precision, it'll by limited due to the fact that the player's level and magic level is a integer, i.e. it will not cover this precision.
```lua
for i = min, max, 0.001 do
    print(i) -- each value incremented by 0.001, covering all values with 3 digits of precision
end
```

And this tell us that every time a user decide to add the damage condition using this list type, he should know (if he wants the proper behavior of this formula) that he will need perform that same operation I did in all monster spells that uses this formula ``math.random(math.floor(min) * 1000, math.floor(max) * 1000) / 1000`` and that he should not apply ``floor`` to it.